speechbrain/nnet/CNN.py

```diff
@@ -1161,3 +1597,78 @@ def get_padding_elem_transposed(
         - 1
     )
     return int(padding)
+
+
+def gabor_impulse_response(t, center, fwhm):
+    denominator = 1.0 / (torch.sqrt(torch.tensor(2.0) * math.pi) * fwhm)
+    gaussian = torch.exp(
+        torch.tensordot(
+            1.0 / (2.0 * fwhm.unsqueeze(1) ** 2),
+            (-(t ** 2.0)).unsqueeze(0),
+            dims=1,
+        )
+    )
+    center_frequency_complex = center.type(torch.complex64)
+    t_complex = t.type(torch.complex64)
+    sinusoid = torch.exp(
+        torch.complex(torch.tensor(0.0), torch.tensor(1.0))
+        * torch.tensordot(
+            center_frequency_complex.unsqueeze(1),
+            t_complex.unsqueeze(0),
+            dims=1,
+        )
+    )
+    denominator = denominator.type(torch.complex64).unsqueeze(1)
+    gaussian = gaussian.type(torch.complex64)
+    return denominator * sinusoid * gaussian
+
+
+def gabor_impulse_response_legacy_complex(t, center, fwhm):
```

Comment:
Legacy ?

Reply:
Yes. Leaf internally has some complex dtype operations, and I used to face problems with these operations in prior versions of torch (as well as in torch-xla, which to my best knowledge still doesn't support grad on those ops on TPUs). _legacy_complex is basically doing these operations as two float tensors instead of a complex dtype tensor.

They can be removed if you like, and there's even a simple flag that can be used to switch, but some people who might need to use an prev torch version (say <=1.9) for different reasons might find this extremely helpful. Your call.

Reply:
Any updates on this?

Reply:
All good.